drv/transceivers-server/src/udp.rs

```diff
@@ -217,397 +215,588 @@ impl ServerImpl {
     fn handle_message(
         &mut self,
         msg: Message,
+        msg_id: u64,
         data: &[u8],
         tx_data_buf: &mut [u8],
     ) -> Option<u32> {
         // If the version is below the minimum committed, then we can't really
         // trust the message, even though it nominally deserialized correctly.
         // Don't reply.
-        if msg.header.version < version::MIN {
-            ringbuf_entry!(Trace::WrongVersion(msg.header.version));
+        if msg.version() < version::inner::MIN {
+            ringbuf_entry!(Trace::WrongVersion(msg.version()));
             return None;
         }
 
-        let (reply, data_len) = match msg.body {
+        let reserved_framing = Header::MAX_SIZE + Message::MAX_SIZE;
+        let (body, data_len) = match msg.body {
             // These messages should never be sent to us, and we reply
-            // with a `ProtocolError` below.
+            // with a `WrongMessage` below.
             MessageBody::SpRequest(..) => {
                 ringbuf_entry!(Trace::GotSpRequest);
-                (SpResponse::Error(Error::ProtocolError), 0)
+                (MessageBody::Error(ProtocolError::WrongMessage), 0)
             }
             MessageBody::SpResponse(..) => {
                 ringbuf_entry!(Trace::GotSpResponse);
-                (SpResponse::Error(Error::ProtocolError), 0)
+                (MessageBody::Error(ProtocolError::WrongMessage), 0)
             }
             // Nothing implemented yet
             MessageBody::HostResponse(r) => {
                 ringbuf_entry!(Trace::UnexpectedHostResponse(r));
                 return None;
             }
             // Happy path: the host is asking something of us!
-            MessageBody::HostRequest(h) => {
-                match self.handle_host_request(
-                    h,
-                    msg.modules,
-                    data,
-                    &mut tx_data_buf[Message::MAX_SIZE..],
-                ) {
-                    Ok(r) => r,
-                    Err(e) => (SpResponse::Error(e), 0),
-                }
+            MessageBody::HostRequest(h) => self.handle_host_request(
+                h,
+                data,
+                &mut tx_data_buf[reserved_framing..],
+            ),
+            // Nothing implemented yet
+            MessageBody::Error(e) => {
+                ringbuf_entry!(Trace::GotError(e));
+                return None;
```

Comment:
Dumb question: why does this branch (and `HostResponse` above)return `None` isntead of `WrongMessage`?

Reply:
There are no dumb questions, Matt ;). In the case of `MessageBody::Error`, a `WrongMessage` would not be the correct response since 1) receiving an `MessageBody::Error` message will be perfectly reasonable once we actually start using `MessageBody::SpRequest` and 2) we don't respond to `Error` messages. A similar line of logic applies to `MessageBody::HostResponse`. That will be a valid message type once we implement `SpRequest`s.

I think the `MessageBody::HostResponse` branch should be changed to respond with `ProtocolError::NotSupported` in the interim, but I don't think we should respond in the `MessageBody::Error` branch.

@bnaecker FYI

Reply:
I'm going to opt to keep both the `HostResponse` and `Error` branches the way they are right now (log to ringbuf, do not respond). Our current philosophy has been "a request gets a response", so responding to a response-class of message would break that. Additionally, responding to an error could cause additional errors, causing us to spiral.